Scripts/Runtime/Entities/TaskSystem/AbstractTaskDriver.cs

```diff
@@ -53,34 +56,40 @@ public abstract class AbstractTaskDriver : AbstractAnvilBase
         /// <summary>
         /// Reference to the associated <see cref="AbstractTaskSystem"/>
         /// </summary>
-        public AbstractTaskSystem TaskSystem { get; }
+        //TODO: #100 - Might be able to change this back so we can access from the outside
+        internal AbstractTaskSystem TaskSystem { get; }
 
         /// <summary>
         /// Reference to the associated <see cref="World"/>
         /// </summary>
         public World World { get; }
 
-        internal CancelRequestsDataStream CancelRequestsDataStream { get; }
-        internal List<AbstractTaskStream> TaskStreams { get; }
-        internal TaskDriverCancellationPropagator CancellationPropagator { get; private set; }
+        internal List<AbstractTaskDriver> SubTaskDrivers { get; }
+        
+        internal TaskDriverCancelFlow CancelFlow { get; }
+        internal TaskData TaskData { get; }
+        internal AbstractTaskDriver Parent { get; }
 
-        protected AbstractTaskDriver(World world, Type systemType)
+        protected AbstractTaskDriver(World world, Type systemType, AbstractTaskDriver parent)
         {
             //We can't just pull this off the System because we might have triggered it's creation via
             //world.GetOrCreateSystem and it's OnCreate hasn't occured yet so it's World is still null.
             World = world;
+            Parent = parent;
 
             TaskSystem = (AbstractTaskSystem)world.GetOrCreateSystem(systemType);
             Context = TaskSystem.RegisterTaskDriver(this);
 
-            m_SubTaskDrivers = new List<AbstractTaskDriver>();
-            TaskStreams = new List<AbstractTaskStream>();
+            SubTaskDrivers = new List<AbstractTaskDriver>();
             m_JobConfigs = new List<AbstractJobConfig>();
 
-            TaskStreamFactory.CreateTaskStreams(this, TaskStreams);
-            CancelRequestsDataStream = new CancelRequestsDataStream();
-
-            TaskDriverFactory.CreateSubTaskDrivers(this, m_SubTaskDrivers);
+            TaskData = new TaskData(this, TaskSystem);
+            CancelFlow = new TaskDriverCancelFlow(this, Parent?.CancelFlow);
+            TaskDriverFactory.CreateSubTaskDrivers(this, SubTaskDrivers);
+            
+            //MIKE - Ordering Question, not sure if there is a better flow
+            CancelFlow.BuildRequestData();
```

Comment:
This is now fixed in my recent work. 

Before this was needed because there was a pass to build down the tree to create all the SubTaskDrivers and it's data but then we had to crawl back up to setup the links of request data and whether things were cancellable. 

Today it's a build phase in the constructor, and then a Harden phase for being ready for job execution and is much cleaner and easier to follow.